guidelines/dxp/metal-plugins/metal-aop.md

```diff
@@ -0,0 +1,70 @@
+# Metal AOP guide of use
+
+> **NOTE:** This plugin was originally in metal-plugins but now lives in portal so we can provide a better developin experience.
+
+AOP provides tools to improve control on what is executed before and after other method is invoked. You can finde this component in `frontend-js-web` module in portal.
+
+## Preparing the use
+
+First of all the `frontend-js-web` dependency shall be added to the `package.json` of the module in which we'll use AOP. Be sure that the version matches with the one in `frontend-js-web/bnd.bnd` file.
+
+```
+	"dependencies": {
+		...
+		"frontend-js-web": "4.0.0",
+        ...
+	},
+```
+
+Once dependency is added AOP shall be imported in the js module:
+
+`import {AOP} from 'frontend-js-web';`
+
+## Use case for AOP.after
+
+Example found in `layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorDragDrop.es.js`
+
+```
+_getRegionInterceptor =
```

Comment:
We can probably design a simpler use case to highlight this. For example, in this case, it would be something like:

// Use case: Change the return value of an instance method invocation
```
import {AOP} from `frontend-js-web`;

const foo = new Foo();
console.log(foo.bar()); // "bar"

const _interceptedFoo = AOP.after(target => {
        return AOP.alterReturn("modified bar")
    }, foo, 'bar');

console.log(foo.bar()); // "modified bar"